crates/ruff/src/rules/flake8_newspaper_style/rules/check_function_calls.rs

```diff
@@ -0,0 +1,171 @@
+use rustpython_parser::ast::{Expr, ExprKind, Located, StmtKind};
+
+use crate::checkers::ast::Checker;
+use crate::registry::AsRule;
+use ruff_diagnostics::{Diagnostic, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::scope::{Binding, BindingKind, ClassDef, FunctionDef, Scope, ScopeKind};
+use ruff_python_ast::types::{Range, RefEquality};
+
+#[violation]
+pub struct FuncDefinedAbove {
+    pub function_name: String,
+    pub function_def_line: usize,
+}
+
+impl Violation for FuncDefinedAbove {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!(
+            "newspaper style: function {} defined in line {} should be moved down.",
+            self.function_name, self.function_def_line
+        )
+    }
+}
+
+/// NEW100
+pub fn check_function_calls(checker: &mut Checker, func: &Expr) {
+    if let Some(diagnostic) = match_callee_expr(checker, func) {
+        if !checker.settings.rules.enabled(diagnostic.kind.rule()) {
+            return;
+        }
+
+        checker.diagnostics.push(diagnostic);
+    }
+}
+
+fn match_callee_expr(checker: &Checker, func: &Expr) -> Option<Diagnostic> {
+    let kind = &checker.ctx.scope().kind;
+    if let ScopeKind::Function(FunctionDef {
+        name: caller_name, ..
+    }) = *kind
+    {
+        let Located { node, .. } = func;
+        if let Some(scope) = checker.ctx.parent_scope() {
+            match scope {
+                Scope {
+                    // caller is defined in class
+                    kind:
+                        ScopeKind::Class(
```

Comment:
Should we be catching nested functions? Like this?

```py
def foo():

    def bop():
        bar()

    def bar():
        x = 1

    def baz():
        x = 1
```

Right now, neither this implementation nor `flake8-newspaper-style` catch this.

Reply:
Done, however we might have to think about how it is possible to completely generalize this problem. E.g. catching nested functions is done recursively, so the nested level shouldn't matter but nested classes are not caught atm.